openauto/CMakeLists.txt

```diff
@@ -0,0 +1,117 @@
+add_library(openauto SHARED
+        App.cpp
+        Service/BluetoothService.cpp
```

Comment:
It's not necessary to explicitly enumerate all of the files.

Reply:
For example: https://stackoverflow.com/questions/57355474/how-to-add-library-with-pattern-for-sources

Reply:
https://cmake.org/cmake/help/v3.14/command/file.html?highlight=file#filesystem

> Note We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The CONFIGURE_DEPENDS flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if CONFIGURE_DEPENDS works reliably, there is still a cost to perform the check on every rebuild.

Reply:
Fair enough. I wasn't necessarily advocating that particular method. I was just suggesting that having to maintain a list of "all files in a directory" is kind of silly. If there is a way to do it with a regex or just by adding by directory name that reduces dependencies and makes the code easier to read.